### ISSUE TITLE
Automatic type conversion for whole number floats

### ISSUE THREAD

**Issue by JFoederer** [robotframework/robotframework#4960]
Type conversions are a very convenient feature in Robot framework. To make them that extra bit convenient I propose an enhancement.

Currently passing any string representation of a `float` number to a keyword accepting only `int` will fail. In most cases this is justified, but there are situations where floats are convertible to `int` just fine. Examples are `"1.0"`, `"2.00"` or `"1e100"`. Note that these conversions currently are already accepted when passed as type `float` (i.e. `${1.0}` or `${1e100}`.  Conversion for numbers for which the decimal part is non-zero should still fail. We are talking about conversion here, not type casting.

**Comment by pekkaklarck** [comment]
Yes, this makes sense. From user perspective it's odd we accept the ${2.0}` but not `2.0`. This is so easy to change we can still do it in RF 7.0. As a bonus this helps mitigating problems caused by changes to internal conversion APIs that some libraries have used.
